portal/src/main/scala/cromwell/pipeline/datastorage/dto/User.scala

```diff
@@ -1,36 +1,76 @@
 package cromwell.pipeline.datastorage.dto
 
-import cromwell.pipeline.datastorage.dto.User.UserEmail
-import play.api.libs.json.{ Json, OFormat }
-import play.api.libs.json.Format
+import cats.data.{ NonEmptyChain, Validated }
+import cromwell.pipeline.utils.validator.Wrapped
+import play.api.libs.json.{ Format, Json, OFormat }
 import slick.lifted.MappedTo
-import play.api.libs.json.Reads._
-import play.api.libs.functional.syntax._
 
 final case class User(
-  userId: UserId,
+  userId: UUID,
   email: UserEmail,
   passwordHash: String,
   passwordSalt: String,
-  firstName: String,
-  lastName: String,
+  firstName: Name,
+  lastName: Name,
   profilePicture: Option[ProfilePicture] = None,
   active: Boolean = true
 )
 
 object User {
   implicit lazy val userFormat: OFormat[User] = Json.format[User]
-  type UserEmail = String
-}
-
-final case class UserId(value: String) extends MappedTo[String]
-
-object UserId {
-  implicit lazy val userIdFormat: Format[UserId] = implicitly[Format[String]].inmap(UserId.apply, _.value)
 }
 
 final case class ProfilePicture(value: Array[Byte]) extends MappedTo[Array[Byte]]
 
 object ProfilePicture {
   implicit lazy val profilePictureFormat: OFormat[ProfilePicture] = Json.format[ProfilePicture]
 }
+
+final class UserEmail private (override val unwrap: String) extends AnyVal with Wrapped[String]
+
+object UserEmail extends Wrapped.Companion {
+  type Type = String
+  type Wrapper = UserEmail
+  type Error = String
+  implicit lazy val emailFormat: Format[UserEmail] = Json.valueFormat[UserEmail]
+  override protected def create(value: String): UserEmail = new UserEmail(value)
+  override protected def validate(value: String): ValidationResult[String] = Validated.cond(
+    value.matches("^[^@]+@[^\\.]+\\..+$"),
+    value,
+    NonEmptyChain.one("Email should match the following pattern <text_1>@<text_2>.<text_3>")
+  )
+}
+
+final class Name private (override val unwrap: String) extends AnyVal with Wrapped[String]
+
+object Name extends Wrapped.Companion {
+  type Type = String
+  type Wrapper = Name
+  type Error = String
+  implicit lazy val nameFormat: Format[Name] = Json.valueFormat[Name]
+  override protected def create(value: String): Name = new Name(value)
+  override protected def validate(value: String): ValidationResult[String] =
+    Validated.cond(
+      value.matches("^[a-zA-Z]+$"),
+      value,
+      NonEmptyChain.one("Name cannot contain spaces, numbers or special characters.sbt.")
```

Comment:
`.sbt.`?

Reply:
Redefine message. Name should contain only letters (but make it more beautiful, please)